errbot/backends/slack.py

```diff
@@ -601,7 +601,8 @@ def _member_joined_channel_event_handler(self, event):
         """Event handler for the 'member_joined_channel' event"""
         user = SlackPerson(self.sc, event["user"])
         if user == self.bot_identifier:
-            self.callback_room_joined(SlackRoom(channelid=event["channel"], bot=self))
+            user = self.bot_identifier
```

Comment:
I don't think this if condition is needed, if the callback is moved from it.
Setting user to self.bot_identifier does nothing from what i can tell.

I don't know offhand what the value of user is, if it does not match `self.bot_identifier`.

Reply:
Slack Backend may generate additional calls when loading Person info, so instead of returning a new SlackPerson, I'm giving the existing one.
Actually I may just compare against `bot_identifier.userid` to make it quicker.
But the main reason for this is to lower the amount of API calls done to Slack.